libs/pageserver_api/src/config.rs

```diff
@@ -296,7 +296,13 @@ pub mod defaults {
 
     pub const DEFAULT_INGEST_BATCH_SIZE: u64 = 100;
 
-    pub const DEFAULT_MAX_VECTORED_READ_BYTES: usize = 128 * 1024; // 128 KiB
+    // Soft limit for the maximum size of a vectored read.
+    // This is determined by the largest NeonWalRecord that can exist (minus dbdir and reldir keys
+    // which are unbounded). As of this writing, that is a `NeonWalRecord::ClogSetCommitted` record,
+    // with 32k xids. That's the max number of XIDS on a single CLOG page. The size of such a record
+    // is `sizeof(Transactionid) * 32768 + (some fixed overhead from 'timestamp`, the Vec length and whatever extra serde serialization adds)`.
+    // That is, slightly above 128 kB.
```

Comment:
```suggestion
    /// Soft limit for the maximum size of a vectored read.
    ///
    /// This is determined by the largest NeonWalRecord that can exist (minus dbdir and reldir keys
    /// which are unbounded). As of this writing, that is a `NeonWalRecord::ClogSetCommitted` record,
    /// with 32k xids. That's the max number of XIDS on a single CLOG page. The size of such a record
    /// is `sizeof(Transactionid) * 32768 + (some fixed overhead from 'timestamp`, the Vec length and whatever extra serde serialization adds)`.
    /// That is, slightly above 128 kB.
```

Make the first line into "short summary", make comment into doc comment.

If it was only made into a doc comment, it would trigger a clippy lint about the short summary line.

Reply:
https://github.com/neondatabase/neon/pull/9221/commits/1464ba15e4cfe5ad39139a4e76818d024ac923d0